sync/src/relayer/transaction_hashes_process.rs

```diff
@@ -0,0 +1,106 @@
+use crate::relayer::compact_block::TransactionHashes;
+use crate::relayer::Relayer;
+use ckb_core::transaction::ProposalShortId;
+use ckb_logger::{debug_target, trace_target};
+use ckb_network::PeerIndex;
+use ckb_protocol::RelayTransactionHashes as FbsRelayTransactionHashes;
+use ckb_store::ChainStore;
+use failure::Error as FailureError;
+use std::convert::TryInto;
+use std::mem;
+
+pub struct TransactionHashesProcess<'a, CS> {
+    message: &'a FbsRelayTransactionHashes<'a>,
+    relayer: &'a Relayer<CS>,
+    peer: PeerIndex,
+}
+
+impl<'a, CS: ChainStore + 'static> TransactionHashesProcess<'a, CS> {
+    pub fn new(
+        message: &'a FbsRelayTransactionHashes,
+        relayer: &'a Relayer<CS>,
+        peer: PeerIndex,
+    ) -> Self {
+        TransactionHashesProcess {
+            message,
+            relayer,
+            peer,
+        }
+    }
+
+    pub fn execute(self) -> Result<(), FailureError> {
+        let transaction_hashes: TransactionHashes = (*self.message).try_into()?;
+
+        let mut x_hashes = Vec::with_capacity(transaction_hashes.hashes.len());
+        {
+            let tx_filter = self.relayer.shared().tx_filter();
+            for tx_hash in transaction_hashes.hashes {
```

Comment:
Can we find a better name for `x_hashes`? naming is hard...
Should use `filter_map` here:

```suggestion
    let x_hashes: Vec<&H256> = transaction_hashes.hashes.iter().filter_map(|tx_hash| {
       ...
    }).collect();
```

Reply:
After changing to `filter_map`, Clippy suggested changing to `filter`.

thanks.